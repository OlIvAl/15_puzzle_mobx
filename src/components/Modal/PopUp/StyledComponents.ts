import styled from 'styled-components';

export const StyledPopUpWrapper = styled.div`
  width: 300px;
  padding: 20px;
  background-color: #fff;
  display: flex;
  flex-direction: column;
  align-items: center;
`;
export const StyledPopUpHeader = styled.div`
  font-size: 24px;
  font-weight: bold;
  margin-bottom: 12px;
`;
export const StyledPopUpContent = styled.div`
  
`;
export const StyledPopUpCloseButton = styled.button`
    border-radius: 4px;
    padding: 6px 10px;
    background-color: #358ada;
    color: #fff;
    text-decoration: none;
    border: none;
    width: 50%;
    margin-top: 20px;
  `;