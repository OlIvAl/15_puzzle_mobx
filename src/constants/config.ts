export const BOARD_TILE_SIZE: number = 4;

export const TILE_SIZE: number = 100;
export const TILE_MARGIN: number = -1;
export const BORD_BORDER: number = 3;

export const BORD_SIZE: number = (TILE_SIZE * BOARD_TILE_SIZE) + (TILE_MARGIN * (BOARD_TILE_SIZE - 1));