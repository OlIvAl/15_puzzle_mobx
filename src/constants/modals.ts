export const WIN_MODAL = 'WIN_MODAL';