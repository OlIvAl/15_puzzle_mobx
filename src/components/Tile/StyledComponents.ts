import styled from 'styled-components';
import {BORD_SIZE, TILE_SIZE} from '../../constants/config';

interface IStyledProps {
  top: number;
  left: number;
}

export const StyledTile = styled('div')<IStyledProps>`
  cursor: pointer;
  background-color: #fff;
  color: #5f5f5f;
  border: 1px solid #aaa;
  display: flex;
  align-items: center;
  justify-content: center;
  font-size: 36px;
  box-sizing: border-box;
  width: ${TILE_SIZE}px;
  height: ${TILE_SIZE}px;
  position: absolute;
  top: ${({top}: IStyledProps): number => top}px;
  left: ${({left}: IStyledProps): number => left}px;
  
  @media (hover: hover) {
    &:hover {
      background: #f5f5f5;
    }
  }
  
  @media (max-width: ${BORD_SIZE * 1.2}px) {
    width: ${TILE_SIZE / 1.5}px;
    height: ${TILE_SIZE / 1.5}px;
  }
`;