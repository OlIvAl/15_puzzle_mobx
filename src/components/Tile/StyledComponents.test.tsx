import * as React from 'react';
import { StyledTile } from './StyledComponents';
import {BORD_SIZE, TILE_SIZE} from '../../constants/config';
import renderer from 'react-test-renderer';

describe('StyledTile', () => {
  const TOP: number = 100;
  const LEFT: number = 200;

  it('он отображается', () => {
    const tree = renderer.create(
        <StyledTile
          top={TOP}
          left={LEFT}
        />
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('имеет заданную ширину', () => {
    const tree = renderer.create(<StyledTile top={TOP} left={LEFT} />).toJSON();

    expect(tree).toHaveStyleRule('width', `${TILE_SIZE}px`);
  });

  it('имеет заданную высоту', () => {
    const tree = renderer.create(<StyledTile top={TOP} left={LEFT} />).toJSON();

    expect(tree).toHaveStyleRule('height', `${TILE_SIZE}px`);
  });

  it('имеет заданную позицию top', () => {
    const tree = renderer.create(<StyledTile top={TOP} left={LEFT} />).toJSON();

    expect(tree).toHaveStyleRule('top', `${TOP}px`);
  });

  it('имеет заданную позицию left', () => {
    const tree = renderer.create(<StyledTile top={TOP} left={LEFT} />).toJSON();

    expect(tree).toHaveStyleRule('left', `${LEFT}px`);
  });

  it('имеет заданную ширину в мобильном виде', () => {
    const tree = renderer.create(<StyledTile top={TOP} left={LEFT} />).toJSON();

    expect(tree).toHaveStyleRule(
      'width',
      `${TILE_SIZE / 1.5}px`,
      {
        media: `(max-width:${BORD_SIZE * 1.2}px)`
      }
    );
  });

  it('имеет заданную высоту в мобильном виде', () => {
    const tree = renderer.create(<StyledTile top={TOP} left={LEFT} />).toJSON();

    expect(tree).toHaveStyleRule(
      'height',
      `${TILE_SIZE / 1.5}px`,
      {
        media: `(max-width:${BORD_SIZE * 1.2}px)`
      }
    );
  });
});