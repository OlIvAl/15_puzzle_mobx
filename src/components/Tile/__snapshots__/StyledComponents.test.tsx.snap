// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledTile он отображается 1`] = `
.c0 {
  cursor: pointer;
  background-color: #fff;
  color: #5f5f5f;
  border: 1px solid #aaa;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 36px;
  box-sizing: border-box;
  width: 100px;
  height: 100px;
  position: absolute;
  top: 100px;
  left: 200px;
}

@media (hover:hover) {
  .c0:hover {
    background: #f5f5f5;
  }
}

@media (max-width:476.4px) {
  .c0 {
    width: 66.66666666666667px;
    height: 66.66666666666667px;
  }
}

<div
  className="c0"
/>
`;
